platforms/scripts/afu_synth_setup_std

```diff
@@ -0,0 +1,251 @@
+#!/usr/bin/env python
+# Copyright(c) 2013-2017, Intel Corporation
+#
+# Redistribution  and  use  in source  and  binary  forms,  with  or  without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of  source code  must retain the  above copyright notice,
+#   this list of conditions and the following disclaimer.
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+# * Neither the name  of Intel Corporation  nor the names of its contributors
+#   may be used to  endorse or promote  products derived  from this  software
+#   without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,  BUT NOT LIMITED TO,  THE
+# IMPLIED WARRANTIES OF  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED.  IN NO EVENT  SHALL THE COPYRIGHT OWNER  OR CONTRIBUTORS BE
+# LIABLE  FOR  ANY  DIRECT,  INDIRECT,  INCIDENTAL,  SPECIAL,  EXEMPLARY,  OR
+# CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+# SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+# INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+# CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+#
+# Create a Quartus build environment in a new directory and configure it for
+# building an AFU.  Run "afu_synth_setup_std --help" for details.
+#
+# This standard version of afu_synth_setup may be used by multiple FPGA
+# platform releases.  Typically, a platform release will provide an
+# afu_synth_setup in the release's bin directory that is a stub.  The stub
+# sets FPGA_RELEASE_DIR and invokes this common script.
+#
+
+import sys
+import os
+import stat
+import errno
+import shutil
+import subprocess
+from os.path import dirname, realpath, sep
+import subprocess
+
+
+def errorExit(msg):
+    sys.stderr.write('Error: {0}\n'.format(msg))
+    sys.exit(1)
+
+
+# The root directory of a system release.
+def getSysReleasePath(args):
+    if (args.release is not None):
+        release_dir = args.release
+    elif ('FPGA_RELEASE_DIR' in os.environ):
+        release_dir = os.environ['FPGA_RELEASE_DIR']
+    else:
+        # Parent directory of the running script
+        release_dir = os.path.dirname(
+            os.path.dirname(os.path.realpath(sys.argv[0])))
+
+    # Confirm that the path looks reasonable
+    if (not os.path.exists(os.path.join(release_dir,
+                                        'hw/lib/fme-ifc-id.txt'))):
```

Comment:
I assume this file is guaranteed to exist?

Reply:
Yes. The packager depends on it, so it's a reasonable test that the user has pointed at a tree that makes sense.